docs/src/description/declarations.md

```diff
@@ -72,3 +86,27 @@ In the above example, `k0` declares a periodic column with a cycle of length `4`
 Periodic columns can be referenced by [integrity constraints](./constraints.md#integrity_constraints) by using the column's identifier.
 
 When constraints are evaluated, these periodic values always refer to the value of the column in the current row. For example, when evaluating an integrity constraint such as `enf k0 * a = 0`, `k0` would be evaluated as `0` in rows `0`, `1`, `2` of the trace and as `1` in row `3`, and then the cycle would repeat. Attempting to refer to the "next" row of a periodic column, such as by `k0'`, is invalid and will cause a `ParseError`.
+
+## Random values (`random_values`)
+
+A `random_values` section contains declarations for random values provided by the verifier. Random values can be accessed by the named identifier for the whole array or by named bindings to single or grouped random values within the array.
+
+**Random values are optional.** However if the section is declared then it cannot be empty and it can only contain a single declaration.
+
+The following is an example of a valid `random_values` source section:
+
+```
+random_values:
+    rand: [4]
+```
+
+where `rand` is a random value array of length `4`. Random values can be accessed using `$` followed by the name of the array and index. For example, `$rand[2]` would refer to the third random value in the array.
```

Comment:
nit: Let's make this a full sentence and clarify the second sentence slightly:

"In the above example, `rand` is a random value array of length `4`. Random values can be accessed using `$` followed by the name of the array and the index of the value. For example, `$rand[2]` would refer to the third random value in the array."